include/Bezier/utils.h

```diff
@@ -0,0 +1,103 @@
+#ifndef UTILS_H
+#define UTILS_H
+
+#include <Eigen/Dense>
+#include <vector>
+
+#include "declarations.h"
+
+#ifndef __cpp_lib_make_unique
+#include <memory>
+namespace std
+{
+template <typename T, typename... Args> inline std::unique_ptr<T> make_unique(Args&&... args)
+{
+  return std::unique_ptr<T>(new T(std::forward<Args>(args)...));
+}
+} // namespace std
+#endif
```

Comment:
In C++11 (which current Bezier supports) doesn't have `std::make_unique`, and we use unique ptr for caching. When moving to c++17, this will be done by `std::optional`